Update PWA Kit to work with the new SLAS channel_id & shopper context siteId change (@W-16253008@)

adamraya opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

The PR consumes the changes in SLAS channel_id and shopper context siteId by bumping commerce-sdk-isomorphic version to the latest.



# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Bump to commerce-sdk-isomorphic@3.0.0

# How to Test-Drive This PR

1. Checkout this branch `W-16253008-update-pwa-kit-with-channel_id-and-siteId`.
2. Enable and [configure the Private client](https://developer.salesforce.com/docs/commerce/pwa-kit-managed-runtime/guide/use-a-slas-private-client.html#configure-the-pwa-kit):
    2.1. `template-retail-react-app/config/default.js` 
    ```
    clientId: '8c7ad811-647f-48ae-9e51-992bd7ea33b3',
    ```
 
    2.2 `template-retail-react-app/app/ssr.js`
    ```
    useSLASPrivateClient: true
    ```
  
    2.3 `template-retail-react-app/app/components/_app-config/index.jsx`
    ```
    enablePWAKitPrivateClient={true}
    ```

3. Start the retail-react-app with the secret private client: `PWA_KIT_SLAS_CLIENT_SECRET=<value> npm start`.
4. Verify in the browser network tab that all SLAS requests get 200 responses.


# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

adamraya committed:
Bump commerce-sdk-isomophic to latest

vcua-mobify commented on packages/template-retail-react-app/app/components/_app-config/index.jsx:
```diff
@@ -68,7 +68,8 @@ const AppConfig = ({children, locals = {}}) => {
             // Make sure to also enable useSLASPrivateClient in ssr.js when enabling this setting.
-            // enablePWAKitPrivateClient={true}
+            // TODO: Changed for testing private client. Restore config before merging.
+            enablePWAKitPrivateClient={true}
```
You've already made a note here, but yes let's undo this change (and the one in ssr.js and default.js) before merging into develop.

The only change we are merging in is the commerce-sdk-isomorphic version update in package.json

adamraya committed:
Fix test

adamraya committed:
Clean up

joeluong-sfcc reviewed (commented):
Nice work testing, glad to see that this didn't require many changes to get working

Few questions though:
1. Will there be a follow up PR for v2 to update the docs?
2. Double checking, the retail-react-app does not use Shopper Context and therefore we do not need to update anything with the siteId changes?
3. Currently for the `commerce-sdk-isomorphic`, in order to avoid a breaking change with siteId being marked as required in the future, we added documentation section about the siteId requirement in the README/CHANGELOG. We stated that its marked as optional for backward compabitibility but will be marked as mandatory by Jan 2025, and when that change happens, we'll only do a minor version bump. **Do we want to do the same thing here to avoid another breaking change?**

adamraya commented:
> Nice work testing, glad to see that this didn't require many changes to get working
> 
> Few questions though:
> 
> 1. Will there be a follow up PR for v2 to update the docs?

Correct. For v2 we'll document the upgrade by creating a separate PR in the docs repo.

> 2. Double checking, the retail-react-app does not use Shopper Context and therefore we do not need to update anything with the siteId changes?

Correct. PWA Kit is not using Shopper Context so is not affected by the siteId changes.

> 3. Currently for the `commerce-sdk-isomorphic`, in order to avoid a breaking change with siteId being marked as required in the future, we added documentation section about the siteId requirement in the README/CHANGELOG. We stated that its marked as optional for backward compabitibility but will be marked as mandatory by Jan 2025, and when that change happens, we'll only do a minor version bump. **Do we want to do the same thing here to avoid another breaking change?**

Sounds good. I added the same comments in the README/CHANGELOG.

adamraya committed:
Add docs

joeluong-sfcc commented on packages/commerce-sdk-react/CHANGELOG.md:
```diff
@@ -1,5 +1,20 @@
+Please be aware that existing tenants are on a temporary allow list and will see no immediate disruption to service.  We do ask that all users seek to adhere to the `channel_id` requirement before the end of August to enhance your security posture before the holiday peak season.
+
+In practice, we recommend:
+- For customers using the SLAS helpers with a private client, it is recommended to upgrade to `v3.0.0` of the `commerce-sdk-react`.
```
Thanks for updating the docs! But question about SemVer:

So the `commerce-sdk-react` consumes the new `commerce-sdk-isomorphic@3.0.0` which contains a breaking change where it now requires `channel_id` with SLAS private client. 

There were no breaking changes needed to consume this new version of the isomorphic SDK since we have `siteId` in the `clientConfig` by default, but PWA Kit customers who consume this new version may break depending on how they've modified their project. 

When we release the new version of `commerce-sdk-react` should it be `v3.0.0` or `v2.1.0`?

adamraya replied:
It should be `commerce-sdk-react@3.0.0` since it includes a breaking change.

adamraya committed:
Merge branch 'develop' into W-16253008-update-pwa-kit-with-channel_id-and-siteId

vcua-mobify reviewed (approved):
Going to approve but minor nit - we probably will want the warning to be at the very top of the list in the changelog.

Also, can we improve the visibility of that message so that it isn't just buried in a changelog that is nested several layers deep in our monorepo? We can do this in a follow up PR I think.

adamraya commented:
> Going to approve but minor nit - we probably will want the warning to be at the very top of the list in the changelog.
> 
> Also, can we improve the visibility of that message so that it isn't just buried in a changelog that is nested several layers deep in our monorepo? We can do this in a follow up PR I think.

I think it's enough as it is now. We show the message in the CHANGELOG and at the top of the README file.
